data-tool/flows/tombstone/tombstone_queries.py

```diff
@@ -1,5 +1,88 @@
+
+def get_unprocessed_corps_subquery(flow_name, environment):
+    subqueries = [
+        {
+            'name': 'default(all corps)',
+            'cte': '',
+            'where': ''
+        },
+        {
+            'name':'TING',
+            'cte': """
+                    with ting_corps as (
+                        select distinct ting_corp_num
+                        from corp_involved_amalgamating
+                    ),
+                    ted_corps as (
+                        select distinct ted_corp_num
+                        from corp_involved_amalgamating
+                    )
+            """,
+            'where': """
+                    and exists (
+                        select 1 from ting_corps t where t.ting_corp_num = c.corp_num
+                    )
+                    and not exists (
+                        select 1 from ted_corps t where t.ted_corp_num = c.corp_num
+                    )
+            """
+        },
+        {
+            'name':'TED that all its TINGs(XP excluded) have been migrated',
+            'cte': f"""
+                with t2 as (
+                    select distinct cia1.ted_corp_num
+                    from corp_involved_amalgamating cia1
+                    where not exists (
+                        select 1
+                        from corp_involved_amalgamating cia2
+                        left join corp_processing cp
+                            on cia2.ting_corp_num = cp.corp_num
+                            and cp.flow_name = '{flow_name}'
+                            and cp.environment = '{environment}'
+                            and cp.processed_status = 'COMPLETED'
+                        where cia2.ted_corp_num = cia1.ted_corp_num
+                        and (cia2.ting_corp_num like 'BC%' or cia2.ting_corp_num like 'Q%' or cia2.ting_corp_num like 'C%')
+                        and cp.corp_num is null
+                    )
+                )
+            """,
+            'where': """
+                and exists (
+                    select 1 from t2 where c.corp_num = t2.ted_corp_num
+                ) 
+            """
+        },
+        {
+            'name':'Other corps, non-TING and non-TED',
+            'cte': """
+                with t3 as (
+                    select ting_corp_num as corp_num
+                    from corp_involved_amalgamating
+                    union
+                    select ted_corp_num as corp_num
+                    from corp_involved_amalgamating
+                )
+            """,
+            'where': """
+                and not exists (
+                    select 1
+                    from t3
+                    where t3.corp_num = c.corp_num
+                )
+            """
+        }
+    ]
+    # change index to select subset of corps
+    # 0: all, 1: TING, 2: TED that linked TINGs are migrated, 3: exclude TING & TED
+    subquery = subqueries[2]
+    return subquery['cte'], subquery['where']
```

Comment:
so with this approach, when we do the actual migration in prod, we would run in the following order? 1, 2, 3 or 1, 2, 0?

Reply:
1, 2, 3 or 2, 1, 3. But you may find fewer eligible corps using the second order. 
0 is for other purpose so that we can tweak queries easily.

Reply:
can you add a comment similar to what you wrote here in the code ^^